api/src/constructor.ts

```diff
@@ -1,13 +1,17 @@
 import { Proxmox } from "./model";
 import ProxmoxEngine from "./ProxmoxEngine";
-import { buildApiProxy } from "./proxy";
+import { ApiRequestable, buildApiProxy } from "./proxy";
 import { ProxmoxEngineOptions } from './ProxmoxEngine';
 
-export function proxmoxApi(options: ProxmoxEngineOptions | ProxmoxEngine): Proxmox.Api {
-    if ((options as any).getTicket)
-        return buildApiProxy(options as ProxmoxEngine, '/api2/json');
+function isProxmoxEngine(val: any): val is ApiRequestable {
+    return 'doRequest' in val;
```

Comment:
maybe it's better to use `typeof val.doRequest === 'function'`?

Reply:
I will check that later.

Reply:
if need server to test ty this
host: https://node1.cepatcloud.id:8006
user: fahri372a@pve
pass: `gxCqEKQYR433iunp`